xbmc/network/WebServer.cpp

```diff
@@ -827,112 +822,98 @@ MHD_RESULT CWebServer::CreateFileDownloadResponse(
     mimeType = CreateMimeTypeFromExtension(ext.c_str());
   }
 
-  if (request.method != HEAD)
-  {
-    uint64_t totalLength = 0;
-    std::unique_ptr<HttpFileDownloadContext> context(new HttpFileDownloadContext());
-    context->file = file;
-    context->contentType = mimeType;
-    context->boundaryWritten = false;
-    context->writePosition = 0;
-
-    if (handler->IsRequestRanged())
-    {
-      if (!request.ranges.IsEmpty())
-        context->ranges = request.ranges;
-      else
-        HTTPRequestHandlerUtils::GetRequestedRanges(request.connection, fileLength,
-                                                    context->ranges);
-    }
+  uint64_t totalLength = 0;
+  std::unique_ptr<HttpFileDownloadContext> context(new HttpFileDownloadContext());
```

Comment:
would be nice to update this to use make_unique but not required.
```suggestion
  std::unique_ptr<HttpFileDownloadContext> context = std::make_unique<HttpFileDownloadContext>();
```

Reply:
Applied.